electricitymap/contrib/parsers/OPENNEM.py

```diff
@@ -533,11 +500,94 @@ def _fetch_au_nsw_au_vic_exchange(
     return datetimes_and_netflows
 
 
+def _build_network_url(
+    path: str,
+    network_code: str,
+    metrics: list[str],
+    target_datetime: datetime,
+    network_region: str,
+) -> str:
+    target_datetime_local = target_datetime.astimezone(ZoneInfo("Australia/Sydney"))
+    target_datetime_naive = target_datetime_local.replace(tzinfo=None)
+
+    date_start = (target_datetime_naive - REFETCH_FREQUENCY).strftime(
+        "%Y-%m-%dT%H:%M:%S"
+    )
+    date_end = target_datetime_naive.strftime("%Y-%m-%dT%H:%M:%S")
+
+    params = [("metrics", m) for m in metrics]
+    params += [
+        ("date_start", date_start),
+        ("date_end", date_end),
+        ("network_region", network_region),
+    ]
+
+    query = urlencode(params)
+    base = f"https://api.openelectricity.org.au/v4/{path}/network/{network_code}"
+
+    return f"{base}?{query}"
+
+
+def _fetch_network_datasets(
+    zone_key: str,
+    session: Session,
+    dataset_type: str,
+    target_datetime: datetime,
+    metrics: list[str],
+) -> list[dict[str, Any]]:
+    network_region = ZONE_KEY_TO_REGION.get(zone_key)
+    network_code = ZONE_KEY_TO_NETWORK.get(zone_key)
```

Comment:
Network code can be null here which cause type issues further down when it expects it to be a string instead of `str | None`